subprojects/solver-z3/src/test/java/hu/bme/mit/theta/solver/z3/Z3SolverTest.java

```diff
@@ -105,7 +110,26 @@ public void testFunc() {
 		final Expr<FuncType<IntType, IntType>> val = optVal.get();
 
 		// Assert
-		assertTrue(val.getType().equals(ca.getType()));
+		assertEquals(ca.getType(), val.getType());
+	}
+	
+	@Test
+	public void testArray() {
+		final Solver solver = Z3SolverFactory.getInstace().createSolver();
+
+        final ConstDecl<ArrayType<IntType, IntType>> arr = Const("arr", Array(Int(), Int()));
+
+        solver.add(ArrayExprs.Eq(Write(arr.getRef(), Int(0), Int(1)), arr.getRef()));
+        solver.add(ArrayExprs.Eq(Write(arr.getRef(), Int(1), Int(2)), arr.getRef()));
+
+        // Check, the expression should be satisfiable
+        SolverStatus status = solver.check();
+        assertTrue(status.isSat());
+        
+        Valuation valuation = solver.getModel();
+        final Optional<LitExpr<ArrayType<IntType, IntType>>> optVal = valuation.eval(arr);
+		final Expr<ArrayType<IntType, IntType>> val = optVal.get();
+		assertEquals(arr.getType(), val.getType());
```

Comment:
As far as I see, this only tests that the types match (Int index and Int values). Could you add assertions to test the literals inside the array, i.e., is element 0 really Int(1) and is element 1 really Int(2)?